app/models/test_track/split_registry.rb

```diff
@@ -0,0 +1,39 @@
+class TestTrack::SplitRegistry
+  def self.from_remote
+    new(TestTrack::Remote::SplitRegistry.to_hash)
+  end
+
+  def initialize(registry_hash)
+    @registry_hash = registry_hash
+  end
+
+  def include?(split_name)
+    registry_hash.key?(split_name)
+  end
+
+  def loaded?
+    registry_hash.present?
+  end
+
+  def split_names
+    registry_hash['splits'].keys
+  end
+
+  def experience_sampling_weight
+    registry_hash.fetch('experience_sampling_weight')
+  end
+
+  def weights_for(split_name)
+    registry_hash && registry_hash['splits'][split_name] && registry_hash['splits'][split_name]['weights']
```

Comment:
TODO: freeze the weights hash to be a little safer

Reply:
like we could freeze the entire registry_hash on ingest, I figure, right?

Reply:
(deep freeze)